File: sample-code/python/test/test_ios_selectors.py
import pytest
import os
import copy

from appium import webdriver
from helpers import report_to_sauce, take_screenshot_and_syslog, IOS_BASE_CAPS, EXECUTOR


class TestIOSSelectors():

    @pytest.fixture(scope='function')
    def driver(self, request, device_logger):
        calling_request = request._pyfuncitem.name

        caps = copy.copy(IOS_BASE_CAPS)
        caps['name'] = calling_request

        driver = webdriver.Remote(
            command_executor=EXECUTOR,
            desired_capabilities=caps
        )

        def fin():
            report_to_sauce(driver.session_id)
            take_screenshot_and_syslog(driver, device_logger, calling_request)
            driver.quit()

        request.addfinalizer(fin)

        driver.implicitly_wait(10)
        return driver

    def test_should_find_elements_by_accessibility_id(self, driver):
        search_parameters_element = driver.find_elements_by_accessibility_id('ComputeSumButton')
        assert 1 == len(search_parameters_element)

    def test_should_find_elements_by_class_name(self, driver):
        window_elements = driver.find_elements_by_class_name('XCUIElementTypeWindow')
        assert 2 == len(window_elements)

    def test_should_find_elements_by_nspredicate(self, driver):
        all_visible_elements = driver.find_elements_by_ios_predicate('visible = 1')
        assert 24 <= len(all_visible_elements)

    def test_should_find_elements_by_class_chain(self, driver):
        window_element = driver.find_elements_by_ios_class_chain('XCUIElementTypeWindow[1]/*')
        assert 1 == len(window_element)

    def test_should_find_elements_by_xpath(self, driver):
        action_bar_container_elements = driver.find_elements_by_xpath('//XCUIElementTypeWindow//XCUIElementTypeButton')
        assert 8 == len(action_bar_container_elements)

File: sample-code/csharp/test/TestIOSSelectors.cs
using NUnit.Framework;
using OpenQA.Selenium;
using OpenQA.Selenium.Appium;
using OpenQA.Selenium.Appium.Enums;
using OpenQA.Selenium.Appium.iOS;
using OpenQA.Selenium.Remote;
using System;
using System.Collections.Generic;
using AppiumDotNetSamples.Helper;

namespace AppiumDotNetSamples
{
    [TestFixture()]
    public class IOSSelectorsTest
    {
        private IOSDriver<IOSElement> driver;

        [OneTimeSetUp()]
        public void BeforeAll()
        {
            DesiredCapabilities capabilities = new DesiredCapabilities();
            capabilities.SetCapability(MobileCapabilityType.BrowserName, "");
            capabilities.SetCapability(MobileCapabilityType.PlatformName, "iOS");
            capabilities.SetCapability(MobileCapabilityType.PlatformVersion, App.IOSPlatformVersion());
            capabilities.SetCapability(MobileCapabilityType.AutomationName, "XCUITest");
            capabilities.SetCapability(MobileCapabilityType.DeviceName, App.IOSDeviceName());
            capabilities.SetCapability(MobileCapabilityType.App, App.IOSApp());

            driver = new IOSDriver<IOSElement> (Env.ServerUri(), capabilities, Env.INIT_TIMEOUT_SEC);
            driver.Manage().Timeouts().ImplicitWait = Env.IMPLICIT_TIMEOUT_SEC;
        }

        [OneTimeTearDown()]
        public void AfterAll()
        {
            driver.Quit();
        }

        [Test()]
        public void TestShouldFindElementsByAccessibilityId()
        {
            ICollection<IOSElement> elements = driver.FindElementsByAccessibilityId("ComputeSumButton");
            Assert.AreEqual(1, elements.Count);
        }

        [Test()]
        public void TestShouldFindElementsByClassName()
        {
            ICollection<IOSElement> elements = driver.FindElementsByClassName("XCUIElementTypeWindow");
            Assert.AreEqual(2, elements.Count);
        }

        [Test()]
        public void TestShouldFindElementsByPredicate()
        {
            // need newer version to use FindElementsByNSPredicate
            ICollection<IOSElement> elements = driver.FindElements("-ios predicate string", "visible = 1");
            Assert.AreEqual(27, elements.Count);
        }

        [Test()]
        public void TestShouldFindElementsByClassChain()
        {
            // need newer version to use FindElementsByClassChain
            ICollection<IOSElement> elements = driver.FindElements("-ios class chain", "XCUIElementTypeWindow[1]/*[2]");
            Assert.AreEqual(1, elements.Count);
        }

        [Test()]
        public void TestShouldFindElementsByXPath()
        {
            ICollection<IOSElement> elements = driver.FindElementsByXPath("//XCUIElementTypeWindow//XCUIElementTypeButton");
            Assert.AreEqual(8, elements.Count);
        }
    }
}
